libs/timeserieslib/SyntheticTimeSeriesCreator.h

```diff
@@ -0,0 +1,116 @@
+#ifndef SYNTHETICTIMESERIESCREATOR_H
+#define SYNTHETICTIMESERIESCREATOR_H
+
+#include "TimeSeriesCsvWriter.h"
+#include "TimeSeries.h"
+#include "McptConfigurationFileReader.h"
+#include "TimeSeriesCsvReader.h"
+
+
+namespace mkc_timeseries
+{
+    template <class Decimal>
+    class SyntheticTimeSeriesCreator : public TimeSeriesCsvReader<Decimal>
+    {
+        public:
+            SyntheticTimeSeriesCreator(const std::string& fileName, 
+                            TimeFrame::Duration timeFrame,
+                            TradingVolume::VolumeUnit unitsOfVolume,
+                            const Decimal& minimumTick) : 
+                mTimeFrame(timeFrame),
+                mUnitsOfVolume(unitsOfVolume),
+                mMinimumTick(minimumTick),
+                TimeSeriesCsvReader<Decimal> (fileName, timeFrame, unitsOfVolume, minimumTick),
```

Comment:
Call base class constructor TimeSeriesCsvReader before the member initialization list. So move this before the initialization of mTimeFrame